smart/plugin_test.go

```diff
@@ -279,7 +279,7 @@ func TestCollectMetrics(t *testing.T) {
 
 			ReadSmartData = func(device string,
 				sysutilProvider SysutilProvider) (*SmartValues, error) {
-				return &SmartValues{}, nil
+				return nil, errors.New("x not valid disk")
```

Comment:
In fact as the next snippet was removed in a previous commit, I had to adapt the code so that the tests pass